pkg/cmd/step/create/step_create_task.go

```diff
@@ -176,6 +177,7 @@ func NewCmdStepCreateTaskAndOption(commonOpts *opts.CommonOptions) (*cobra.Comma
 	cmd.Flags().BoolVarP(&options.ViewSteps, "view", "", false, "Just view the steps that would be created")
 	cmd.Flags().BoolVarP(&options.EffectivePipeline, "effective-pipeline", "", false, "Just view the effective pipeline definition that would be created")
 	cmd.Flags().BoolVarP(&options.SemanticRelease, "semantic-release", "", false, "Enable semantic releases")
+	cmd.Flags().BoolVarP(&options.UseBranchAsRevision, "branch-as-revision", "", false, "Use the provided branch as the revision for release pipelines, not the version tag")
```

Comment:
Not quite sure the intent of the flag becomes clear here. 

So why is this even a release pipeline if it is not supposed to be doing release stuff? Does this not come down to the fact that we are missing something conceptually, namely the ability to run anything else than a release or pr pipeline?

Reply:
Yup, that's the situation.